warp/yul/ToCairoVisitor.py

```diff
@@ -210,181 +153,65 @@ def visit_expression_statement(self, node: ast.ExpressionStatement) -> str:
-
+            return f"let ({vars_repr}) = {value_repr}"
         else:
-            self.preamble = True
-            vars_repr = self.generate_ids_typed(node.variables)
+            assert len(node.variables) == 1
             return f"{vars_repr} = {value_repr}"
 
     def visit_block(self, node: ast.Block) -> str:
-        return "\n".join(self.print(x) for x in node.statements)
-
-    def to_high_low_felt(self, vars: str) -> str:
-        if vars != "":
-            vars = vars.replace(" : Uint256", "")
-            vars = vars.split(",")
-            params_repr = ", ".join(f"{p.strip()}_low, {p.strip()}_high" for p in vars)
-            return params_repr
-        else:
-            return vars
-
-    def init_u256(self, params_repr: str) -> str:
-        params = params_repr.replace(" : Uint256", "")
-        params = params.split(",")
-        declr_str = ""
-        for p in params:
-            if "local" in p:
-                p_no_local = p.replace("local", "").strip()
-                declr_str += (
-                    f"{p}: Uint256 = Uint256({p_no_local}_low, {p_no_local}_high)\n"
-                )
-            else:
-                declr_str += f"local {p}: Uint256 = Uint256({p}_low, {p}_high)\n"
-        return declr_str
-
-    def set_ret_vars_len(self, node: ast.FunctionDefinition):
-        if node.name in self.public_functions:
-            # must split into high/low bits
-            self.current_function_ret_vars_len = len(node.return_variables) * 2
-            self.curr_fun_ret_felt = True
-        else:
-            self.current_function_ret_vars_len = len(node.return_variables)
-            self.curr_fun_ret_felt = False
-
-    def gen_wrapped_extern_assign(self, fun_name: str, params: str, return_repr: str) -> str:
-        if self.current_function_ret_vars_len == 0:
-            return f"""{fun_name}{{range_check_ptr=range_check_ptr,
-            pedersen_ptr=pedersen_ptr,
-            storage_ptr=storage_ptr,
-            memory_dict=memory_dict,
-            msize=msize}}({params})
-            return ()"""
-        else:
-            return f"""let ({return_repr}) = {fun_name}{{range_check_ptr=range_check_ptr,
-            pedersen_ptr=pedersen_ptr,
-            storage_ptr=storage_ptr,
-            memory_dict=memory_dict,
-            msize=msize}}({params})
-            return ({return_repr})"""
+        stmt_reprs = []
+        for stmt in node.statements:
+            with self._new_statement():
+                stmt_reprs.append(self.visit(stmt))
+                if self.last_needs_ref_copy:
+                    stmt_reprs.append(gen_ref_copy())
+        return "\n".join(stmt_reprs)
 
     def visit_function_definition(self, node: ast.FunctionDefinition):
         self.last_function = node
-        self.set_ret_vars_len(node)
-        taboos = ["abi_", "checked_", "getter_", "panic_error"]
+        taboos = ["abi_", "checked_add", "getter_"]
```

Comment:
Why did we remove panic_error

Reply:
`RevertNormalizer` automatically removes all "reverting" functions and turns calls to them to reverts. "reverting" functions are functions such that each execution path of them leads to a revert

Reply:
So, that automatically removes "panic_error"